src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterFactory.cs

```diff
@@ -19,7 +19,7 @@ public abstract class JsonConverterFactory : JsonConverter
         /// </summary>
         protected JsonConverterFactory() { }
 
-        internal JsonConverter GetConverterInternal(Type typeToConvert, JsonSerializerOptions options)
+        internal JsonConverter GetConverterInternal(Type typeToConvert, JsonSerializerOptions? options)
```

Comment:
I believe this can return null. @steveharter - can you please confirm?
```suggestion
        internal JsonConverter? GetConverterInternal(Type typeToConvert, JsonSerializerOptions? options)
```

Reply:
The reason made it non null: all overrides returning Activator.CreateInstance(...) which wouldn't return null for ref types, and some of calling routine throwing on null return

Reply:
> and some of calling routine throwing on null return

Exactly. Since the callers of `GetConverterInternal` need to check for null and validate/throw if it is, the contract of this method should be that it **can** return null.